cmd/server/main.go

```diff
@@ -61,8 +78,58 @@ func main() {
 	}()
 
 	if err := server.ListenAndServe(); err != nil && !errors.Is(err, http.ErrServerClosed) {
-		log.Fatal().Err(err).Msg("Error starting server")
+		log.Error().Err(err).Msg("Error starting server")
+		cancel()
 	}
 
 	log.Info().Msg("Server shut down")
 }
+
+func getDB(connectionString string) (*gorm.DB, error) {
```

Comment:
[CHANGE REQUEST] Please simplify the `main` package.

Reply:
small refactoring